packages/notifi-dapp-example/src/components/TopicOptions.tsx

```diff
@@ -183,19 +210,33 @@ export const TopicOptions = <T extends TopicRowCategory>(
                   }
                 } else {
                   if (
-                    isUserInputValid(
-                      props.userInputParam.kind,
-                      evt.target.value,
-                    )
+                    lowerBound &&
+                    upperBound &&
+                    (Number(evt.target.value) > upperBound ||
+                      (Number(evt.target.value) < lowerBound &&
+                        evt.target.value !== ''))
                   ) {
-                    selectOrInputValue(evt.target.value);
+                    setErrorMessage(
+                      `Please enter a value between ${lowerBound} and ${upperBound}`,
+                    );
+                  } else {
+                    if (
+                      isUserInputValid(
+                        props.userInputParam.kind,
+                        evt.target.value,
+                        props.userInputParam.customInputConstraints,
+                      )
+                    ) {
+                      selectOrInputValue(evt.target.value);
+                    }
```

Comment:
Not sure if we still need the following hardcode section:

```ts
if (props.userInputParam.options[0]) {
                  if (
                    Number(evt.target.value) > 99 ||
                    (Number(evt.target.value) < 1 && evt.target.value !== '')
                  ) {
                    setErrorMessage('Please enter a value between 1 and 99');
                  }
                } else {
```

Reply:
I think we do need the hardcode section, as there is no uppper/lower bound set up for liquidation alert

Reply:
Hmmmm... I think we will eventually remove the condition `props.userInputParam.options[0]`. If it is not proper to do it now, shall we make a `TODO:` comment so with can easily understand this part and upcoming action.